cloud/azure/deploytf/.nitric/modules/stack/main.tf

```diff
@@ -8,7 +8,8 @@ resource "random_string" "stack_id" {
 }
 
 locals {
-  stack_name = "${var.stack_name}-${random_string.stack_id.result}"
+  stack_id   = random_string.stack_id.result
+  stack_name = "${var.stack_name}-${local.stack_id}"
```

Comment:
This isn't necessarily an issue with this PR in particular but more a source of confusion in general: `stack_name` is ambiguously named because it both refers to the name of the stack based on the stack file and, in this case, also the stack file name with the random id. It may be better to disambiguate these definitions across the board so that we don't run into similar bugs down the line.